src/plugins/discover/public/application/main/services/load_saved_search.ts

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+import type { SavedSearch } from '@kbn/saved-search-plugin/public';
+import { cloneDeep, isEqual } from 'lodash';
+import { loadAndResolveDataView } from '../utils/resolve_data_view';
+import { DiscoverInternalStateContainer } from './discover_internal_state_container';
+import { DiscoverDataStateContainer } from './discover_data_state_container';
+import { cleanupUrlState } from '../utils/cleanup_url_state';
+import { getValidFilters } from '../../../utils/get_valid_filters';
+import { DiscoverStateContainer, LoadParams } from './discover_state';
+import { addLog } from '../../../utils/add_log';
+import { DiscoverSavedSearchContainer } from './discover_saved_search_container';
+import {
+  DiscoverAppState,
+  DiscoverAppStateContainer,
+  getInitialState,
+} from './discover_app_state_container';
+import { DiscoverServices } from '../../../build_services';
+
+interface LoadSavedSearchDeps {
+  appStateContainer: DiscoverAppStateContainer;
+  dataStateContainer: DiscoverDataStateContainer;
+  internalStateContainer: DiscoverInternalStateContainer;
+  savedSearchContainer: DiscoverSavedSearchContainer;
+  services: DiscoverServices;
+  setDataView: DiscoverStateContainer['actions']['setDataView'];
+}
+
+/**
+ * Loading persisted saved searches or existing ones and updating services accordingly
+ * @param params
+ * @param deps
+ */
+export const loadSavedSearch = async (
+  params: LoadParams,
+  deps: LoadSavedSearchDeps
+): Promise<SavedSearch> => {
+  addLog('[discoverState] loadSavedSearch');
+  const { savedSearchId, useAppState } = params ?? {};
+  const { appStateContainer, internalStateContainer, savedSearchContainer, services } = deps;
+  const appState = useAppState ? appStateContainer.getState() : undefined;
+  // First let's clean up the previous state
+  services.filterManager.setAppFilters([]);
+  services.data.query.queryString.clearQuery();
```

Comment:
We could do that, so basically you're suggesting to switch cleanup & loading ... so in case a saved search loading fails, previous state is not cleaned up?

```
// First let's clean up the previous state
  services.filterManager.setAppFilters([]);
  services.data.query.queryString.clearQuery();
  if (!useAppState) {
    appStateContainer.set({});
  }
  // Then, take care of data view and saved search loading
  let nextSavedSearch = savedSearchId
    ? await savedSearchContainer.load(savedSearchId)
    : await savedSearchContainer.new(
        await getStateDataView(params, { services, appState, internalStateContainer })
      );
```

to

```

  // Take care of saved search loading
  let nextSavedSearch = savedSearchId
    ? await savedSearchContainer.load(savedSearchId)
    : await savedSearchContainer.new(
        await getStateDataView(params, { services, appState, internalStateContainer })
      );

//  Clean up the previous state
  services.filterManager.setAppFilters([]);
  services.data.query.queryString.clearQuery();
  if (!useAppState) {
    appStateContainer.set({});
  }
```

Reply:
> so in case a saved search loading fails, previous state is not cleaned up?

Yes

Reply:
makes sense, even if it doesn't make a difference here since if it fails, going back to a previous state wouldn't work, but it make sense for future iterations

Reply:
seems functionals didn't like this change 😅  ... need to research why

Reply:
So, before the last changes `savedSearchContainer.new` was still using the queryString & filters of the data services, that's why functionals broke. Now it works, and the order of executions (1 load, 2 reset) works